lib/CodeTransform/Adapter/TolerantParser/Refactor/TolerantExtractExpression.php

```diff
@@ -93,7 +93,7 @@ function (Node $item) use ($endNode) {
     
     private function getCommonExpression(Node $node1, Node $node2): ?Expression
     {
-        if ($node1 == $node2 && $node1 instanceof Expression) {
+        if ($node1 === $node2 && $node1 instanceof Expression) {
```

Comment:
might fix the "nesting level too deep" issue: https://stackoverflow.com/questions/3834791/fatal-error-nesting-level-too-deep-recursive-dependency